cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/CaReconciler.java

```diff
@@ -225,64 +234,79 @@ Future<Void> reconcileCas(Clock clock) {
-
-                    clientsCa = new ClientsCa(reconciliation, certManager,
-                            passwordGenerator, clientsCaCertName,
-                            clientsCaCertSecret, clientsCaKeyName,
-                            clientsCaKeySecret,
+                            generateClusterCa,
+                            clusterCaConfig != null ? clusterCaConfig.getCertificateExpirationPolicy() : null);
+
+
+                    clientsCa = new ClientsCa(reconciliation, certManager, passwordGenerator,
+                            clientsCaCertName, existingClientsCaCertSecret,
+                            clientsCaKeyName, existingClientsCaKeySecret,
                             ModelUtils.getCertificateValidity(clientsCaConfig),
                             ModelUtils.getRenewalDays(clientsCaConfig),
-                            clientsCaConfig == null || clientsCaConfig.isGenerateCertificateAuthority(),
+                            generateClientsCa,
                             clientsCaConfig != null ? clientsCaConfig.getCertificateExpirationPolicy() : null);
-                    clientsCa.createRenewOrReplace(reconciliation.namespace(),
-                            caLabels, Map.of(), Map.of(),
-                            clientsCaConfig != null && !clientsCaConfig.isGenerateSecretOwnerReference() ? null : ownerRef,
-                            Util.isMaintenanceTimeWindowsSatisfied(reconciliation, maintenanceWindows, clock.instant()));
-
-                    Promise<Void> caUpdatePromise = Promise.promise();
 
                     List<Future<ReconcileResult<Secret>>> secretReconciliations = new ArrayList<>(2);
 
-                    if (clusterCaConfig == null || clusterCaConfig.isGenerateCertificateAuthority())   {
-                        Future<ReconcileResult<Secret>> clusterSecretReconciliation = secretOperator.reconcile(reconciliation, reconciliation.namespace(), clusterCaCertName, clusterCa.caCertSecret())
-                                .compose(ignored -> secretOperator.reconcile(reconciliation, reconciliation.namespace(), clusterCaKeyName, clusterCa.caKeySecret()));
+                    if (generateClusterCa)   {
+                        clusterCa.createRenewOrReplace(Util.isMaintenanceTimeWindowsSatisfied(reconciliation, maintenanceWindows, clock.instant()),
+                                isForceReplace(existingClusterCaKeySecret),
+                                isForceRenew(existingClusterCaCertSecret));
+
+                        OwnerReference ownerReference = clusterCaConfig != null && !clusterCaConfig.isGenerateSecretOwnerReference() ? null : ownerRef;
+                        clusterCaCertSecret = createCaCertSecret(clusterCaCertName, clusterCaCertLabels, clusterCaCertAnnotations, ownerReference, clusterCa, existingClusterCaCertSecret);
+                        Secret clusterCaKeySecret = createCaKeySecret(clusterCaKeyName, ownerReference, clusterCa, existingClusterCaKeySecret);
```

Comment:
I was wondering if we should do the Secret creation async and joining the future. Right now (but I guess even before) the code was sync on Secrets creation (so blocking the main Vert.x thread).

Reply:
The Secret creation is just creating the class, it doesn't actually call Kubernetes. So I don't think that needs to be async. I did wonder if the `createRenewOrReplace` should be async, as that calls the openssl commands and maybe could take a bit of time, but I didn't want to muddle this PR with that change.